### ISSUE TITLE
Design for picking a specific version

### ISSUE THREAD

**Issue by daveverwer** [SwiftPackageIndex/SwiftPackageIndex-Server#1021]
The more I think about the popover that we currently have in #1, I think it's a less than ideal solution. The UI around "from version" and "to version" doesn't make much sense in the context of a playground.

I see this tool being used much more as "Let me try this out" more than "This is a playground I'm going to persist and upgrade over a long period of time. With that in mind, I'd like to propose a new design for the app.

**Comment by daveverwer** [comment]
The UI should default to this:

<img width="1366" alt="Swift Package Index - 2021-04-08 11 02 50 4" src="https://user-images.githubusercontent.com/5180/114011427-1f555a80-985d-11eb-8521-07b261db01a6.png">

* We should use "repo/owner on GitHub" rather than the URL as that fits better with the package registry changes coming soon.
* The default for the drop-down should be "Latest stable - x.x.x" with the latest stable version.

Then, dropping down that drop-down looks like this:

<img width="1366" alt="Swift Package Index - 2021-04-08 11 02 50 3" src="https://user-images.githubusercontent.com/5180/114011584-4ca20880-985d-11eb-93e6-238e160071b0.png">

Just five options:

* Latest stable, latest beta, default branch are self-explanatory.
* The ellipsis at the end of the last two options show more information will be required. See below.

When "Specific version" is selected:

<img width="1366" alt="Swift Package Index - 2021-04-08 11 02 50 2" src="https://user-images.githubusercontent.com/5180/114011721-78bd8980-985d-11eb-89f2-e8f649dca54d.png">

This second drop-down should show all tags that are semantic version numbers.

Or, if "Specific branch/commit" is selected:

<img width="1366" alt="Swift Package Index - 2021-04-08 11 02 50" src="https://user-images.githubusercontent.com/5180/114011832-97238500-985d-11eb-858d-09e9609c538a.png">

* The second drop-down now shows all branches
* The commit field is a free text field, and is optional. If no commit is specified, it uses the latest commit on the selected branch.

What do you think?

**Comment by finestructure** [comment]
Yes, that all sounds good but I'd definitely want to ship this (or even no editable version at all) to see how/if it's being adopted first.

**Comment by daveverwer** [comment]
OK.

I think we might consider shipping it *only* supporting the latest stable version for now then. When I played with the popover version it was quite inconvenient to use with it all being free text fields. There are also bugs with it, which I didn't log as that's what made me think about the design from a slightly higher level, but for example:

https://user-images.githubusercontent.com/5180/114014583-ad7f1000-9860-11eb-94c7-f37a22f51353.mp4

**Comment by finestructure** [comment]
Yes, that's because the text field are validated but the validation errors aren't being exposed yet. So in that case the edit buffer is being discarded.

The issue with the alpha is that the Version parser doesn't handle the full SemanticVersion spec and therefore is rejected.
